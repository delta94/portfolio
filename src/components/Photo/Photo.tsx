import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { CustomTheme } from '../../theme';

interface PhotoProps {
  url?: string;
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  photo: {
    width: 200,
    height: 150,
    borderRadius: 30,
    padding: 16,
    cursor: 'pointer',
  },
  photoMobile: {
    width: 150,
    height: 114,
  },
}));

const Photo: React.FC<PhotoProps> = ({ url }) => {
  const classes: Record<'photo' | 'photoMobile', string> = useStyles();

  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });
  return (
    <img
      onClick={() => window.open('https://www.instagram.com/sk_izsk/', '_blank')}
      className={clsx([classes.photo, isMobile && classes.photoMobile])}
      alt='photoBanner'
      src={url}
    />
  );
};

export { Photo };
