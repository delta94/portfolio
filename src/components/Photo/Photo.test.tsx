import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { Photo } from './Photo';

test('Photo renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <Photo url='mock url' />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
