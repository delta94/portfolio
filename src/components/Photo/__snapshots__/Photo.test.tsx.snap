// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders correctly 1`] = `
<DocumentFragment>
  <img
    alt="photoBanner"
    class="photo-0-2-1"
    src="mock url"
  />
</DocumentFragment>
`;
